File: 백준/Silver/1439. 뒤집기/뒤집기.py
s = input()
one_cnt = 0  # 0에서 1로 바뀌는 횟수
zero_cnt = 0  # 1에서 0으로 바뀌는 횟수

if s[0] == '1':
    zero_cnt += 1
else:
    one_cnt += 1

for idx in range(len(s) - 1):
    if s[idx] != s[idx + 1]:  # 앞의 값과 다를 때,
        if s[idx + 1] == '1':
            zero_cnt += 1  # 0 -> 1이면 0 += 1,
        else:
            one_cnt += 1  # 1 -> 0이면 1 += 1

# 1과 0을 뒤집는 횟수 중, 낮은 수 출력
print(min(one_cnt, zero_cnt))

File: 백준/Silver/1439. 뒤집기/뒤집기.cc
#include "iostream"

using namespace std;
string s = "";
int zero_cnt, one_cnt = 0;

void init() {
    if (s[0] == '1') zero_cnt++; else one_cnt++;
}

void flip() {
    for (int i = 0; i < s.length() - 1; ++i) {
        if (s[i] != s[i + 1]) {
            if (s[i + 1] == '1') zero_cnt++; else one_cnt++;
        }
    }
}

int main() {
    cin >> s;
    init();
    flip();
    cout << min(zero_cnt, one_cnt);
}